backend/server/controllers/user.js

```diff
@@ -31,8 +32,49 @@ export const getUsersById = async (req, res) => {
   try {
     const { params } = req;
     const user = await models.users.findByPk(params.id);
-    res.status(200).send(user);
+    const authToken = req.get('authToken');
+
+    if (!authToken) {
+      res.status(400).send({
+        "Error message": "Auth token not provided"
+      })
+    } 
+
+    const decodedUser = Authentication.extractUser(authToken);
+
+    if (decodedUser.id != params.id) {
+      res.status(401).send({
```

Comment:
Doesn't this check if the logged in user is the same as the user being viewed? I think logged in users should be able to view any user as long as they are logged in, so as long as ` const decodedUser = Authentication.extractUser(authToken);` does not throw any error, the process should go ahead

Reply:
Yeah, you're right